File: ios/CalendarApp/AppIconManager.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(AppIconManager, NSObject)
RCT_EXTERN_METHOD(changeIcon:(NSString *)iconName resolver:(RCTPromiseResolveBlock)resolver rejecter:(RCTPromiseRejectBlock)rejecter)
RCT_EXTERN_METHOD(getIcon:(RCTPromiseResolveBlock)resolver rejecter:(RCTPromiseRejectBlock)rejecter)
@end

File: ios/CalendarApp/AppIconManager.swift
import UIKit
import React

@objc(AppIconManager)
class AppIconManager: NSObject {
  @objc func changeIcon(_ iconName: String?, resolver: @escaping RCTPromiseResolveBlock, rejecter: @escaping RCTPromiseRejectBlock) {
    DispatchQueue.main.async {
      UIApplication.shared.setAlternateIconName(iconName) { error in
        if let error = error {
          rejecter("ICON_ERROR", error.localizedDescription, error)
        } else {
          resolver(nil)
        }
      }
    }
  }

  @objc func getIcon(_ resolver: RCTPromiseResolveBlock, rejecter: RCTPromiseRejectBlock) {
    resolver(UIApplication.shared.alternateIconName)
  }

  @objc static func requiresMainQueueSetup() -> Bool { return false }
}
